packages/builder/src/steps/pull.ts

```diff
@@ -92,21 +92,29 @@ const pullSpec = {
 
     if (!deployInfo) {
       throw new Error(
-        `deployment not found: ${source}. please make sure it exists for the cannon network and ${preset} preset.`
+        `deployment not found: ${source}. please make sure it exists for the cannon network and ${preset} preset.`,
       );
     }
 
     if (deployInfo.status === 'partial') {
       throw new Error(
-        `deployment status is incomplete for ${source}. cannot generate artifacts safely. please complete deployment to continue import.`
+        `deployment status is incomplete for ${source}. cannot generate artifacts safely. please complete deployment to continue import.`,
       );
     }
 
     return {
       imports: {
         [importLabel]: {
           url: (await runtime.registry.getUrl(source, chainId))!, // todo: duplication
-          ...(await getOutputs(runtime, new ChainDefinition(deployInfo.def), deployInfo.state))!,
+          ...(await getOutputs(
+            runtime,
+            new ChainDefinition(deployInfo.def, false, {
+              chainId,
+              timestamp: deployInfo.timestamp || 0,
+              package: { version: '0.0.0' },
+            }),
```

Comment:
Having to create this dummy object on several places just to be able to instantiate looks prone to error, maybe a helper function to create the ChainDefinition would be useful?

Reply:
Oh, I see that those are overrides, but are they necessary? looks like they are only used for giving them to the access recorder, in that case wouldn't be enough to just send the dummy object? Why is there an option to send custom overrides?